### PR TITLE
bring back go vet

### PR DESCRIPTION
**Pull request by rdhariwal**
## Description

We removed go vet when we upgraded to go 1.12. This PR adds go vet back into pre-commit hooks with an exception for `dutystationloader` package because of a known [issue](https://github.com/golang/go/issues/30465).

## Reviewer Notes
You can test go vet by changing a go file like `server.go` and running `pre-commit run go-vet -a`

## Setup

Checkout branch and run
```sh
pre-commit run go-vet -a 
```

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/164639267) for this change

### CODE REVIEW

File: bin/pre-commit-go-vet [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 set -e
 
-exec 5>&1
-for file in "$@"; do
-    output="$(go vet "$file" 2>&1 | tee /dev/fd/5)"
-    [[ -z "$output" ]]
-done
+for file in $(go list ./... | grep -v /vendor/)
+do
+    # skip analyzing dutystationsloader until repeated tags from embedded types are fixed in Go1.12.2
```
Comment (rdhariwal) [inline bin/pre-commit-go-vet]: See this [link](https://github.com/golang/go/issues/30465) why I am skipping the `dutystationloader` package

File: bin/pre-commit-go-vet [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+    if [[ "$file" == *"dutystationsloader"* ]]; then
+      echo "skipping: $file"
```
Comment (rdhariwal) [inline bin/pre-commit-go-vet]: helpful feedback if script is executed directly

File: bin/pre-commit-go-vet (hunk 1 above)
Comment (rdhariwal) [reply bin/pre-commit-go-vet]: you can repro the issue by running `go vet github.com/transcom/mymove/internal/pkg/dutystationsloader`

**Comment by pjdufour-truss** [comment]
I think you should look to run `go vet` only once and just pass it all the package names, minus the excluded ones.  It might be faster, but I haven't timed it.

```
bash -c 'export go_vet_pkgs=$(go list ./... | grep -v /vendor/ | grep -v internal/pkg/dutystationsloader | tr "\n" " ") && go vet $go_vet_pkgs'
```

**Comment by rdhariwal** [comment]
> 🚀 - would be nice to have our own truss repo with this at some point.

story created: https://www.pivotaltracker.com/story/show/164787341

**Comment by rdhariwal** [comment]
> I think you should look to run `go vet` only once and just pass it all the package names, minus the excluded ones, rather than running go vet multiple times for each package. It might be faster, but I haven't timed it. For example,
> 
> ```
> bash -c 'export go_vet_pkgs=$(go list ./... | grep -v /vendor/ | grep -v internal/pkg/dutystationsloader | tr "\n" " ") && go vet $go_vet_pkgs'
> ```
@pjdufour-truss @chrisgilmerproj 
image
*minor differences* :trollface: 
1st run is using the for loop and the second one is using the one liner from Patrick!

**Comment by rdhariwal** [comment]
@pjdufour-truss I think this is ready for merging, one more pass please.

File: bin/pre-commit-go-vet [hunk 3]
```diff
@@ -1,11 +1,9 @@
...
-exec 5>&1
-for file in "$@"; do
-    output="$(go vet "$file" 2>&1 | tee /dev/fd/5)"
-    [[ -z "$output" ]]
-done
+# skip analyzing dutystationsloader until repeated tags from embedded types are fixed in Go1.12.2
+# https://github.com/golang/go/issues/30465
+go_vet_pkgs=$(go list ./... | grep -v /vendor/ | grep -v internal/pkg/dutystationsloader | tr "\n" " ")
+
+# shellcheck disable=SC2086
```
Comment (rdhariwal) [inline bin/pre-commit-go-vet]: wrapping the variable in `"` converts it to a string that go vet parses as one command which is not something we want, hence the override.

Comment (rdhariwal) [reply bin/pre-commit-go-vet]: ^^ yep that is why I ended up splitting it up a bit so readability is higher.. having said that I am open to either option if you feel strongly about it.

**Pull request merged**